torch_geometric/nn/models/basic_gnn.py

```diff
@@ -435,6 +435,49 @@ def __repr__(self) -> str:
                 f'{self.out_channels}, num_layers={self.num_layers})')
 
 
+def make_batches_cacheable(model: BasicGNN) -> BasicGNN:
+    r"""A decorator overwriting :class:`torch_geometric.nn.models.BasicGNN`
+    inference loop, making it work with
+    :class:`torch_geometric.loader.CachedLoader`.
+    """
+    @torch.no_grad()
+    def inference_with_cache(
+        self,
+        loader: NeighborLoader,
+        device: Optional[Union[str, torch.device]] = None,
+        embedding_device: Union[str, torch.device] = 'cpu',
+        progress_bar: bool = False,
```

Comment:
Wondering why we need this new function here. Can't we simply add a `cache` option to the `inference` call and do `CachedLoader(loader)` if set to `True`?

Reply:
With caching the whole batch ([above comment](https://github.com/pyg-team/pytorch_geometric/pull/7896#discussion_r1297118891)), indeed your suggestion is a better option.

Reply:
Done.

Reply:
Update, Please see [this comment](https://github.com/pyg-team/pytorch_geometric/pull/7896#discussion_r1297570176).